docs/php/activities.md

```diff
@@ -3,3 +3,309 @@ id: activities
+A large execution history can thus adversely impact the performance of your Workflow.
+Therefore, be mindful of the amount of data you transfer via Activity invocation parameters or return values.
+
+Otherwise, no additional limitations exist on Activity implementations.
+
+```php
+class FileProcessingActivitiesImpl implements FileProcessingActivities {
+
+    private S3Client $s3Client;
+
+    private string $localDirectory;
+
+    public function __construct(S3Client $s3Client, string $localDirectory) {
+        $this->s3Client = $s3Client;
+        $this->localDirectory = $localDirectory;
+    }
+
+    public function upload(string $bucketName, string $localName, string $targetName): void
+    {
+        $this->s3Client->putObject(
+            $bucketName,
+            $targetName,
+            fopen($this->localDirectory . $localName, 'rb+')
+        );
+    }
+
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->s3Client->downloadObject(
+            $bucketName,
+            $remoteName,
+            fopen($this->localDirectory .$localName, 'wb+')
+        );
+    }
+
+    public function processFile(string $localName): string
+    {
+        // Implementation omitted for brevity.
+        return compressFile($this->localDirectory . $localName);
+    }
+
+    public function deleteLocalFile(string $fileName): void
+    {
+        unlink($this->localDirectory . $fileName);
+    }
+}
+```
+
+### Accessing Activity Info
+
+The [Activity](https://github.com/temporalio/sdk-php/blob/master/src/Activity.php) class provides static getters to access information about the Workflow that invoked it.
+Note that this information is stored in a thread local variable.
+Therefore, calls to Activity accessors succeed only in the process that invoked the Activity function.
+
+```php
+use Temporal\Activity;
+
+class FileProcessingActivitiesImpl implements FileProcessingActivities
+{
+    // ...
+
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->logger->info("namespace=" +  Activity.getNamespace());
+
+        $info = Activity::getInfo();
+
+        $this->logger->info("workflowId=" + $info->workflowExecution->getID());
+        $this->logger->info("runId=" + $info->workflowExecution->getRunID());
+        $this->logger->info("activityId=" + $info->id);
+        $this->logger->info("activityDeadline=" + $info->deadline);
+
+        // ...
+    }
+
+    // ...
+}
+```
+
+### Activity Heart Beating
+
+Some Activities are long-running.
+To react to a crash quickly, use a heartbeat mechanism.
+`Activity::heartbeat()` lets the Temporal service know that the Activity is still alive.
+You can piggyback `details` on an Activity heartbeat.
+If an Activity times out, the last value of `details` is included in the `TimeoutFailure` delivered to a Workflow.
```

Comment:
I believe it is also available inside the activity through Activity::[getHeartbeatDetails](https://github.com/temporalio/sdk-php/blob/f53f1627f63fca73bd0e533831a8104bacd44309/src/Activity.php#L90). When activity is restarted after failure getHeartbeatDetails allows to learn value from the last successful heartbeat.

Ideally, we should modify the sample to include getHeartbeatDetails.